components/transactions/RecurrencePicker.tsx

```diff
@@ -0,0 +1,121 @@
+    const [recurrenceInstance, setRecurrenceInstance] = React.useState(weekday[0]);
+    const [monthInstance, setMonthInstance] = React.useState(months[0]);
+    let monthEndInstance = monthEndDate[0];
+    
+    const headerText = 20;
+    const bodyText = 16;
+
+    let pickerStyling = {
+        height: 50,
+    }
+    
+    let recurrenceTypePicker = 
+    <Picker style={{width: 120, height: 30, fontSize: bodyText}} selectedValue={recurrenceType} onValueChange={(itemValue) => {setRecurrenceType(itemValue.toString())}} prompt="Recurrence Type">
+        <Picker.Item label="Week(s)" value={weekValue}/> 
+        <Picker.Item label="Month(s)" value={monthValue}/>
+        <Picker.Item label="Year(s)" value={yearValue}/>
+    </Picker>;
+
+    let recurrenceDistanceInput = 
+    <TextInput keyboardType="decimal-pad" 
+        onEndEditing={(event) => {setRecurrenceDistance(_.toInteger(event.nativeEvent.text))}}
+        style={{textAlignVertical: "top", fontSize: bodyText, paddingLeft: 15}}>
+        {recurrenceDistance}
+    </TextInput>;
+
+    let weeklySelector = 
+        <Picker selectedValue={recurrenceInstance} onValueChange={(itemValue) => {setRecurrenceType(itemValue.toString())}} style={pickerStyling}>
+            {weekday.map((instance) => {return <Picker.Item label={instance} value={instance} key={instance}/>})}
+        </Picker>;
+
+    let monthlySelector = 
+        <Picker selectedValue={recurrenceInstance} onValueChange={(itemValue) => {setRecurrenceType(itemValue.toString())}} style={pickerStyling}>
+            {date.map((instance) => {return <Picker.Item label={instance} value={instance} key={instance}/>})}
+        </Picker>;
+
+    
+    //month with 31 days
+    if(monthInstance == months[0] || monthInstance == months[2] || monthInstance == months[4] || monthInstance == months[6] || 
+        monthInstance == months[7] || monthInstance == months[9] || monthInstance == months[11]){
+        monthEndInstance = monthEndDate[0];
+    }
+    else{
+        //February
+        if(monthInstance == months[1]) monthEndInstance = monthEndDate[2];
+        //months with 30 days
+        else monthEndInstance = monthEndDate[1];
+    }
+
+    let yearlySelector =
+        <View style={{flexDirection: "row", height: 50}}>
+            <Picker selectedValue={recurrenceInstance} onValueChange={(itemValue) => {setRecurrenceType(itemValue.toString())}} style={{flex: 0.5}}>
+                {date.slice(0, monthEndInstance).map((instance) => {return <Picker.Item label={instance} value={instance} key={instance}/>})}
+            </Picker>
+            <Picker selectedValue={monthInstance} onValueChange={(itemValue) => {setMonthInstance(itemValue.toString())}} style={{flex: 0.5}}>
+                {months.map((instance) => {return <Picker.Item label={instance} value={instance} key={instance}/>})}
+            </Picker>
+        </View>;
+
+    let recurrenceInstanceSelector;
+
+    switch (recurrenceType) {
```

Comment:
If you can't find a date-time-picker component, then abstract this switch statement into a function and just call that function inside the return component.